src/csvcubed/models/cube/qb/validationerrors.py

```diff
@@ -318,7 +320,119 @@ class EmptyQbMultiMeasureDimensionError(SpecificValidationError):
+        self.message = (
+            "The field attribute of a QbMultiMeasureDimension must be populated"
+        )
+
+
+@dataclass
+class PivotedShapeMeasureColumnsExistError(BothMeasureTypesDefinedError):
+    """
+    An error to inform the user that they have attempted to define a pivoted shape cube with measure columns.
+    """
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "https://gss-cogs.github.io/csvcubed-docs/external/guides/errors/build-command-errors/multiple-measure-columns/"
+
+    def __post_init__(self):
+        self.message = f"The cube is in pivoted shape, but you have defined 1 or more Measure columns. These two approaches are incompatible."
+
+
+@dataclass
+class DuplicateMeasureError(SpecificValidationError):
+    """
+    An error to inform the user that the obesrvation colums contain duplicate values
+    """
+
+    column_names: List[str]
+    column_names_concatenated: str = field(init=False)
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/dup-measure"
+
+    def __post_init__(self):
+        self.column_names_concatenated = ", ".join(self.column_names)
+        self.message = f"In the pivoted shape, two or more observation value columns cannot be represented by identical measures. {self.column_names_concatenated}"
+
+
+@dataclass
+class AttributeNotLinkedError(SpecificValidationError):
+    """
+    An error to inform the user that the units or attribute column is defined but it is not linked to the obs column
+    """
+
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/att-not-linked"
+
+    def __post_init__(self):
+        self.message = f"Units or attribute column '{self.attribute_column_title}' is defined but it is not linked to an observed values column"
+
+
+@dataclass
+class LinkedObsColumnDoesntExistError(SpecificValidationError):
+    """
+    An error to infrom the user that the units or attribute column is defined for which obs val column doesn't appear to exist.
+    """
+
+    alleged_obs_val_column_title: str
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/link-obs-col-not-exist"
+
+    def __post_init__(self):
+        self.message = f"The unit or attribute column '{self.attribute_column_title}' is defined for an observed value column '{self.alleged_obs_val_column_title}' that doesn't appear to exist."
+
+
+@dataclass
+class LinkedToNonObsColumnError(SpecificValidationError):
+    """
+    An error to infrom the user that units or attribute column is defined in which
+    the linked obs val column isn't actually an observations column.
+    """
+
+    alleged_obs_val_column_title: str
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/link-non-obs-col"
```

Comment:
I can't remember. Did we ensure that all of these PURL URLs were correctly configured?